cli/module_build.go

```diff
@@ -1039,26 +1173,66 @@ func reloadModuleActionInner(
 				return err
 			}
 		}
-		if err := addShellService(c, vc, part.Part, true); err != nil {
+
+		// Start the "Reloading to part..." parent step if not already started (for local builds with cloud-built artifacts)
+		if !cloudBuild {
+			if err := pm.Start("reload"); err != nil {
+				return err
+			}
+		}
+		if err := pm.Start("shell"); err != nil {
 			return err
 		}
-		infof(c.App.Writer, "Copying %s to part %s", buildPath, part.Part.Id)
+		shellAdded, err := addShellService(c, vc, logger, part.Part, true)
+		if err != nil {
+			_ = pm.Fail("shell", err)                                //nolint:errcheck
+			_ = pm.FailWithMessage("reload", "Reloading to part...") //nolint:errcheck
+			return err
+		}
+		if shellAdded {
+			if err := pm.CompleteWithMessage("shell", "Shell service installed"); err != nil {
+				return err
+			}
+		} else {
+			if err := pm.CompleteWithMessage("shell", "Shell service already exists"); err != nil {
+				return err
+			}
+		}
+
 		globalArgs, err := getGlobalArgs(c)
 		if err != nil {
 			return err
 		}
 		dest := reloadingDestination(c, manifest)
+
+		if err := pm.Start("upload"); err != nil {
+			return err
+		}
 		err = vc.copyFilesToFqdn(
 			part.Part.Fqdn, globalArgs.Debug, false, false, []string{buildPath},
-			dest, logging.NewLogger(reloadVersionPrefix), args.NoProgress)
```

Comment:
this was the only location where we used `NoProgress` for module reloading. I think ideally we should not create the progress manager if `NoProgress` is true, and have all PM methods return be no-ops if the `pm` pointer is nil. Barring that, we should probably remove `NoProgress` from the args type so we don't try to parse it, and make it hidden (though still existent for backwards compatibility) in the `viam module reload` args list.